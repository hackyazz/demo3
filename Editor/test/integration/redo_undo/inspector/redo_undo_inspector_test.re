open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: inspector", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test simulate set currentSceneTreeNode", () => {
      beforeEach(() => {
        TestTool.closeContractCheck();
        MainEditorSceneTool.initState(~sandbox, ());
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
        );
      });
      afterEach(() => {
        GameObjectTool.clearCurrentSceneTreeNode();
        TestTool.openContractCheck();
      });

      describe(
        "test set currentSceneTreeNode, have two state add into undoStack", () =>
        describe("test undo operate", () => {
          test("test not undo", () =>
            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch
          );
          describe("test undo one step", () =>
            test("step which from second to first", () => {
              RedoUndoTool.undoHistoryState();

              BuildComponentTool.buildInspectorComponent(
                TestTool.buildEmptyAppState(),
                InspectorTool.buildFakeAllShowComponentConfig(),
              )
              |> ReactTestTool.createSnapshotAndMatch;
            })
          );
          describe("test undo two step", () =>
            test("step which from second to zero", () => {
              RedoUndoTool.undoHistoryState();
              RedoUndoTool.undoHistoryState();

              BuildComponentTool.buildInspectorComponent(
                TestTool.buildEmptyAppState(),
                InspectorTool.buildFakeAllShowComponentConfig(),
              )
              |> ReactTestTool.createSnapshotAndMatch;
            })
          );
        })
      );
    });
  });