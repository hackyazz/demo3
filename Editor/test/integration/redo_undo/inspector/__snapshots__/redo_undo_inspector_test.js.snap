// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: inspector test simulate set currentSceneTreeNode test set currentSceneTreeNode, have two state add into undoStack test undo operate test not undo 1`] = `
<article
  className="wonder-inspector-component"
/>
`;

exports[`redo_undo: inspector test simulate set currentSceneTreeNode test set currentSceneTreeNode, have two state add into undoStack test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-inspector-component"
/>
`;

exports[`redo_undo: inspector test simulate set currentSceneTreeNode test set currentSceneTreeNode, have two state add into undoStack test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-inspector-component"
/>
`;
